test/unit/cpu/test-indexsets.cxx

```diff
@@ -37,6 +48,106 @@ TEST_F(IndexSetTest, InvalidSegments)
   EXPECT_NE(true, index_sets_[0].push_back_nocopy(&rs_segment));
 }
 
+TYPED_TEST_P(DependentIndexSetTest, ForwardDependentIndexSets)
+{
+  using ExecPolicy = typename std::tuple_element<0, TypeParam>::type;
+
+  const int numSegments = 10;
```

Comment:
It seems like all of this setup code could go into a `virtual void SetUp()` method in the fixture class. The only differences between the two tests are the cases of `i < numSegments - 1` and `i > 0`, which I think could be easily handled via some virtual helper methods, then just inherit the fixture class to specialize on them. That might be way more work than necessary (I have no problem with you leaving it out) but it might be worth considering.